cluster/src/main/java/org/apache/iotdb/cluster/client/sync/SyncDataHeartbeatClient.java

```diff
@@ -38,14 +38,19 @@
   private SyncDataHeartbeatClient(TProtocolFactory protocolFactory, Node node, SyncClientPool pool)
       throws TTransportException {
     // the difference of the two clients lies in the port
+    // Heartbeat client connection timeout should not be larger than heartbeat interval, otherwise
+    // the thread pool of sending heartbeats or requesting votes may be used up by waiting for
+    // establishing connection with some slow or dead nodes.
     super(
         protocolFactory.getProtocol(
             RpcTransportFactory.INSTANCE.getTransport(
                 new TSocket(
                     TConfigurationConst.defaultTConfiguration,
                     node.getInternalIp(),
                     node.getDataPort() + ClusterUtils.DATA_HEARTBEAT_PORT_OFFSET,
-                    RaftServer.getConnectionTimeoutInMS()))));
+                    Math.min(
```

Comment:
```Math.min(
                        (int) RaftServer.getHeartbeatIntervalMs(),
                        RaftServer.getConnectionTimeoutInMS()))))```
can be saved in `RaftServer` as a new field.

Reply:
+1

Reply:
A new static field is added to RaftServer.